fang/postgres_migrations/migrations/2022-08-20-151615_create_fang_tasks/up.sql

```diff
@@ -3,12 +3,12 @@ CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
 CREATE TYPE fang_task_state AS ENUM ('new', 'in_progress', 'failed', 'finished', 'retried');
 
 CREATE TABLE fang_tasks (
-     id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+     id BYTEA PRIMARY KEY,
```

Comment:
can we provide backward compatible migrations like oban does? 

so changes can be applied sequentially

Reply:
I do not know how to do that.

`BYTEA` makes compatible with `SQLite` because `SQLite` does not have `UUID` type.

Maybe this can be changed to `UUID` if only if `UUID` can be casted to `BYTEA` in postgresql query.

Reply:
I got good news and bad news.


PostgreSQL `uuid` type can not be casted to bytea.

PostgreSQL `uuid` type can be casted to string.

![image](https://github.com/ayrat555/fang/assets/48651252/18bba56b-bab8-4113-ac61-1e1ee06c5af0)

Reply:
Fang PostgreSQL can have the same migrations if we change SQLITE migrations to ID store uuid as `TEXT`, it takes more storage (36 B (TEXT) instead of 16 B (BYTEA)) , but it is worth if you have same migrations from PostgreSQL.

Reply:
text to uuid conversion also works fine.


![image](https://github.com/ayrat555/fang/assets/48651252/a4ffbc27-3525-4ebd-b92a-d9866171ff34)

Reply:
[store uuid as text , so this way PostgreSQL schema will not change](https://github.com/ayrat555/fang/pull/141/commits/5119dcf870a991ef2fcbbdcb0b471dec92739f89)

This comment should address you concern

Reply:
We will do in other PR

Reply:
See issue #142